src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -739,10 +739,11 @@ def __init__(self, **kwargs):
 
     def setup(self):
         webui = self.middleware.call_sync('datastore.query', 'system.settings')
-        if (webui[0]['stg_guiprotocol'] == 'http' or
-           webui[0]['stg_guiprotocol'] == 'httphttps'):
+        if not webui[0]['stg_guihttpsredirect']:
             self.port = int(webui[0]['stg_guiport'] or 80)
-            self.regtype = "_http._tcp."
+        else:
+            self.port = 6000
```

Comment:
This service (`_http._tcp.`) advertises Web UI and RESTful API access. If unencrypted access is not available (disabled with redirect), we should not advertise this service at all.